pems-server/src/main/java/org/maroubra/pemsserver/monitoring/utsapi/WebSensor.java

```diff
@@ -1,80 +1,96 @@
+
+    @Override
+    public String toString() {
+        return "Sensor Family: " + config.getConfig().get("rFamily") + " Sensor: " + config.getConfig().get("rSensor")
+                + " Sub Sensor: " + config.getConfig().get("rSubSensor");
     }
 
-    public void setDatesPollInterval() {
-        fromDate = LocalDateTime.now().minusMinutes(pollIntervalMinutes).withNano(0);
-        toDate = LocalDateTime.now().withNano(0);
+    @Override
+    public boolean start() {
+        timer = new Timer(true);
+        webSensorTask = new WebSensorTask(config,pollIntervalMinutes, sensorLogPublisher, webAPi);
+        timer.schedule(webSensorTask, 0, pollIntervalMinutes * 60 * 1000);
+        return true;
     }
 
-    public void setDates(LocalDateTime fromDate, LocalDateTime toDate) {
-        this.fromDate = fromDate;
-        this.toDate = toDate;
+    @Override
+    public boolean stop() {
+        timer.cancel();
+        return true;
     }
 
+    @Override
+    public Flowable<SensorLog> logs() { return sensorLogPublisher.onBackpressureLatest();  }
+
+    @FactoryClass
+    public interface Factory extends Sensor.Factory<WebSensor> {
+        @Override
+        WebSensor create(@Assisted SensorConfig config);
 
-    private Map<String, String> getQueryParameters() {
-        Map<String, String> parameters = new LinkedHashMap<>();
-        parameters.put("rSubSensor", subSensor);
-        parameters.put("rSensor", sensor);
-        parameters.put("rFamily", family);
-        parameters.put("rToDate", toDate.toString());
-        parameters.put("rFromDate", fromDate.toString());
-        return parameters;
+        @Override
+        Config getConfig();
     }
 
-    public List<String[]> pollSensor() {
-        setDatesPollInterval();
-        List<String[]> data = null;
-        Call<List<String[]>> call = service.getJsonData(getQueryParameters());
-        try {
-            data = call.execute().body();
-        } catch (IOException e) {
-            e.getMessage();
-            log.warn("This sensor may not be returning data please check the sensor api webpage.");
+    @ConfigClass
+    public static class Config implements SensorConfig {
+
+        private String id;
+        private String family;
+        private String subSensor;
+        private String sensor;
+
+        @Override
+        public String getId() { return id; }
+
+        @Override
+        public void setId(String id) {
+            this.id = id;
         }
-        return data;
-    }
 
-    @Override
-    public String toString() {
-        return "Sensor Family: " + family + " Sensor: " + sensor + " Sub Sensor: " + subSensor;
+        @Override
+        public String type() { return WebSensor.class.getCanonicalName(); }
+
+        public Map<String,String> getConfig() {
+            return ImmutableMap.of("rFamily", family, "rSensor", sensor, "rSubSensor", subSensor);
```

Comment:
There's a couple places we use these query strings. Maybe put them into `private static final` variables.